### DISCUSSION TITLE
Strange behaviour between staging and prod

### DISCUSSION THREAD

**Discussion by agenceKanvas** [Pierre-Lannoy/wp-decalog#62] [Q&A]
Hi !
I've just put a website in production. 
The servers are the same (cloudways live/staging), the database is the same, only the urls are modified (of course), and the files are the same. 
But for a reason that I don't get, I have nothing in the console on the live server (when the staging still works perfectly). 
If you have any idea. I probably forgot something but I don't know why ...
Thanks !!

**Comment by Pierre-Lannoy** [comment]
Hello!
That's strange. Are you speaking of the "web console" or the wp-cli tail command? Are the loggers correctly started?

**Comment by agenceKanvas** [comment]
Both ! I've checked them 2 and I don't have any log except for the DecaLog: Live console launched.  
How can I check if the loggers started ? They work well with the same code in staging. 
This is how I init them in the plugin : 
```
add_action( 'after_setup_theme', 'translator_load' );
function translator_load() {
	require_once( plugin_dir_path( __FILE__ ) . 'vendor/autoload.php' );
	\DecaLog\Engine::initPlugin( 'kv-translator', 'Traduction WF', '0.0.1' );
	\DecaLog\Engine::initPlugin( 'kv-translator-ai', 'Traduction AI', '0.0.1' );
```
Maybe I should init decalog with init instead of after_setup_theme ?

**Comment by Pierre-Lannoy** [comment]
OK.
First of all, I can only recommend initializing the DecaLog engine in the init of each plugin, and not in an external function or hook (it is not designed to really work like that): each plugin must have its own engine instance...
Of course this design choice may have an impact. As I haven't tested with your pattern, I can not tell you if it's the cause.
I suggest you to init engine from each plugin first lines, then we will try to understand what's going wrong if it still doesn't work.

**Comment by agenceKanvas** [comment]
I changed it for :
```
add_action( 'init', 'translator_load' );
```
But still the same behaviour. Decalog is working, I have some wordpress logs. But still nothing coming from my plugin.

**Comment by agenceKanvas** [comment]
Oh maybe I misunderstood something. What do you mean by it's own instance ? 
Get it out of even the init hook with something like that ? 
```
require_once( plugin_dir_path( __FILE__ ) . 'vendor/autoload.php' );
\DecaLog\Engine::initPlugin( 'kv-translator', 'Traduction WF', '0.0.1' );

add_action( 'init', 'translator_load' );
function translator_load() {
	global $pagenow;
```

**Comment by Pierre-Lannoy** [comment]
I'm sorry, I don't get it…
If you have a plugin which slug is `kv-translator`, you have to initialize DecaLog engine in the first lines of THIS plugin.

**Comment by agenceKanvas** [comment]
Je vais switcher en français :) 
Le plugin s'appelle bien kv-translator. Je m'étais gourré de hook avec le after_theme_setup, du coup je l'ai remplacé par init

**Comment by Pierre-Lannoy** [comment]
J'entend bien, mais ça ne s'initialise pas dans un hook (en tout cas, ce n'est pas prévu pour ça, peut être que dans certains cas ça peut fonctionner). Ça doit vraiment se faire dans les premières lignes du plugin kv-translator…

**Comment by agenceKanvas** [comment]
Oui pas de soucis, je vais changer ça ! 
Je viens de comprendre une chose. La différence qu'il y a entre mon staging et la prod c'est le plugin de cache. 
Sachant qu'avec un peu de patience (de nombreuses minutes plus tard), après avoir lancé mes actions, j'ai les logs qui s'affichent tous ensemble. Comme si il avait rempli une pile et m'avait tout affiché une fois que tout était terminé.

**Comment by Pierre-Lannoy** [comment]
Si c'est du cache de code ou du cache objet, ça peut avoir un impact au "démarrage".

**Comment by agenceKanvas** [comment]
Ok donc, après des heures de tests et autres modifications, le résultat est sans appel, je pense que le soucis vient du type d'infra que j'utilisais. Cloudways autonomous en l'occurence. 
Sur le staging, avec un seul pod, tout se passait correctement justement parce qu'un seul pod devait être activé pour le staging. En prod autre histoire. 
J'ai migré pour repasser vers une version classique de Cloudways car Polylang était selon le support de Cloudways pas tout à fait compatible avec leur infra, et là miracle, tout se remet à fonctionner.
Merci pour ton aide en tout cas !! Comme toujours :)

**Comment by Pierre-Lannoy** [comment]
Ahhhh. Tant mieux si ça fonctionne maintenant 😅
